store/src/main/java/com/nytimes/android/external/store/base/impl/RealInternalStore.java

```diff
@@ -53,40 +53,79 @@
     RealInternalStore(Fetcher<Raw, Key> fetcher,
                       Persister<Raw, Key> persister,
                       KeyParser<Key, Raw, Parsed> parser,
-                      Cache<Key, Observable<Parsed>> memCache,
                       StalePolicy stalePolicy) {
-        init(fetcher, persister, parser, memCache, stalePolicy);
+        init(fetcher, persister, parser, stalePolicy);
     }
 
-
     RealInternalStore(Fetcher<Raw, Key> fetcher,
                       Persister<Raw, Key> persister,
                       KeyParser<Key, Raw, Parsed> parser,
+                      MemoryPolicy memoryPolicy,
                       StalePolicy stalePolicy) {
-        memCache = CacheBuilder.newBuilder()
-                .maximumSize(getCacheSize())
-                .expireAfterAccess(getCacheTTL(), TimeUnit.SECONDS)
-                .build();
-        init(fetcher, persister, parser, memCache, stalePolicy);
+        init(fetcher, persister, parser, memoryPolicy, stalePolicy);
+    }
 
+    private void init(Fetcher<Raw, Key> fetcher,
+                      Persister<Raw, Key> persister,
+                      KeyParser<Key, Raw, Parsed> parser,
+                      MemoryPolicy memoryPolicy,
+                      StalePolicy stalePolicy) {
+        this.fetcher = fetcher;
+        this.persister = persister;
+        this.parser = parser;
+        this.stalePolicy = stalePolicy;
+
+        initMemCache(memoryPolicy);
+        initFlightRequests(memoryPolicy);
+
+        subject = BehaviorSubject.create();
     }
 
```

Comment:
Minor consideration, since this `init()` is almost the same as the one below, with the exception of creating the `defaultMemoryPolicy`, maybe consolidate the two, adding a conditional to configure a default `MemoryPolicy` if the none was passed in (or `null` passed in).  It may also make sense to move this into a constructor, since nothing else calls `init()`.